tests/core_tests.py

```diff
@@ -332,35 +340,101 @@ def test_queries_endpoint(self):
+        self.assertEquals(4, len(data))
 
         query = db.session.query(models.Query).filter_by(
-            sql='SELECT * FROM ab_user').first()
+            sql='SELECT * FROM ab_user1').first()
         query.changed_on = utils.EPOCH
         db.session.commit()
 
         data = self.get_json_resp('/caravel/queries/{}'.format(123456000))
-        self.assertEquals(1, len(data))
+        self.assertEquals(3, len(data))
 
         self.logout()
         resp = self.client.get('/caravel/queries/{}'.format(0))
         self.assertEquals(403, resp.status_code)
 
-    def test_search_query_endpoint(self):
-        userId = 'userId=null'
-        databaseId = 'databaseId=null'
-        searchText = 'searchText=null'
-        status = 'status=success'
-        params = [userId, databaseId, searchText, status]
-        resp = self.client.get('/caravel/search_queries?'+'&'.join(params))
-        self.assertEquals(200, resp.status_code)
+    def test_search_query_on_db_id(self):
+      self.login('admin')
+      # Test search queries on database Id
+      resp = self.get_resp('/caravel/search_queries?database_id=1')
+      data = json.loads(resp)
+      self.assertEquals(3, len(data))
+      for k, v in data.items():
+        assert data[k]['dbId'] == 1
+
+      resp = self.get_resp('/caravel/search_queries?database_id=-1')
+      data = json.loads(resp)
+      self.assertEquals(0, len(data))
+      self.logout()
+
+    def test_search_query_on_user(self):
+      self.login('admin')
+      # Test search queries on user Id
+      user = appbuilder.sm.find_user('admin')
+      resp = self.get_resp('/caravel/search_queries?user_id={}'.format(user.id))
+      data = json.loads(resp)
+      self.assertEquals(2, len(data))
+      for k, v in data.items():
+          self.assertEquals(v['userId'], user.id)
+
+      user = appbuilder.sm.find_user('gamma')
+      resp = self.get_resp('/caravel/search_queries?user_id={}'.format(user.id))
+      data = json.loads(resp)
+      self.assertEquals(1, len(data))
+      for (k, v) in data.items():
+          self.assertEquals(v['userId'], user.id)
+      self.logout()
+
+    def test_search_query_on_status(self):
+      self.login('admin')
+      # Test search queries on status
+      resp = self.get_resp('/caravel/search_queries?status=success')
+      data = json.loads(resp)
+      self.assertEquals(2, len(data))
+      for k, v in data.items():
+          self.assertEquals(v['state'], "success")
+
+      resp = self.get_resp('/caravel/search_queries?status=failed')
+      data = json.loads(resp)
+      self.assertEquals(1, len(data))
+      for k, v in data.items():
+          self.assertEquals(v['state'], "failed")
+      self.logout()
+
+    def test_search_query_on_text(self):
+      self.login('admin')
+      resp = self.get_resp('/caravel/search_queries?search_text=permission')
+      data = json.loads(resp)
+      self.assertEquals(1, len(data))
+      for k, v in data.items():
```

Comment:
by the way if you don't use the var you can name it _
`for _, v in data.items()`